### ISSUE TITLE
Opening a notification of an organisation while switched into another organisation is broken.

### ISSUE THREAD

**Issue by pulkonet** [zulip/zulip-mobile#2295]
Reproduction steps:
- Log into two organizations
- Switch to first organization and send notification to second organisation (using another client)
-  Click on the notification that comes up
It will lead to this type of screen
![screenshot_20180412-205411](https://user-images.githubusercontent.com/24983958/38700101-997c49f2-3eb7-11e8-827c-a5c8f0a105bb.png)
Tested on OnePlus 3T : Android 8.0.0

**Comment by pulkonet** [comment]
Essentially, clicking on the notification leads to a screen but doesn't switch the organisations.

**Comment by borisyankov** [comment]
Yeah, known issue, but we didn't have a specific issue open to track that. Thanks.

**Comment by kunall17** [comment]
@zulipbot claim

**Comment by gnprice** [comment]
The symptoms of this appear to have changed: @borisyankov investigated recent reports of following a notification leading to the screen being blank white except for the compose box, and determined it was caused by this issue.

**Comment by gnprice** [comment]
In a report today ([chat](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Notifications.20do.20not.20take.20me.20to.20the.20message/near/679848) -- presumably on 21.2.106, the latest release) that sounds like this issue, the symptoms the user describes are
> when I click on [a notification], it often just opens the "home" screen.

That would be an improvement over the behavior described above! Namely
> the screen being blank white except for the compose box

and other symptoms before that. So it's possible that a recent change has made this issue a bit milder.

The fundamentals haven't changed, though.

**Comment by borisyankov** [comment]
The blank screen from the previous report is likely to have been fixed by @jainkuniya work to fix rendering of the `ChatScreen` and have probably reversed to the original behavior.

**Comment by gnprice** [comment]
> The blank screen from the previous report is likely to have been fixed by @jainkuniya work to fix rendering of the `ChatScreen` and have probably reversed to the original behavior.

Ah indeed, thanks for making that connection! That'd be #3089 .

The reported current behavior is better than we had before that bug too, though:
> it seems like the application can't fetch or load any new data. The screens for new unread messages and private messages are stuck [...]

or before that

> It will lead to this type of screen [... message list with "loading" animation ...]

**Comment by gnprice** [comment]
A helpful review on the Play Store summarized the effect of this as meaning they "[c]an't rely on notifications":

> Can't rely on notifications. Doesn't switch to relevant account when opening notification, opens last used account. Without reliable notifications its very stressful and time consuming to cycle through accounts to check for new messages.

**Comment by gnprice** [comment]
One thought I just had: it would be a solid improvement even just to *switch to the right account* when a notification from the non-active account is opened. Even if it just left you at the main "unreads" screen for the account, and didn't navigate to the specific conversation.

That should at least relieve the burden of "cycl[ing] through accounts" to find where a notification-causing message was. And if the account in question doesn't get a high volume of messages (which I suspect it usually won't for people who use lots of different Zulip accounts), then the unreads screen will pretty efficiently get you to the message anyway.

For Android users to get the full benefit of this (or any other solution to this issue), we'll need to also take care of #2691 . At present, because on Android we show a single notification in the UI that aggregates all notification-causing messages, it'll only be possible for this to work when you have just a single notified message at a time, or several all from the same account. But I expect that's the common case anyway.

**Comment by gnprice** [comment]
> One thought I just had: it would be a solid improvement even just to _switch to the right account_ when a notification from the non-active account is opened.

This is done since #3648 landed, and that was released with version 26.11.134. That should mitigate this issue pretty well for most users.

Still open before this issue is properly fixed are the following pieces (quoting from f8d9da85d ):
 * After switching accounts, we stop at the main "unreads" screen;
   we don't go on to narrow to the proper conversation.
 * If the user has several known accounts in the same realm, we
   don't switch to either of them, because we can't tell which.
 * On Android, we show just one notification in the UI even if there
   are messages in several conversations.  Opening it leads to
   whichever message we notified about last; there's no way to use the
   notification to get to the other conversations.  This is #2691.
